appeals/web/src/server/appeals/appeal-details/lpa-questionaire/outcome-incomplete/outcome-incomplete.controller.js

```diff
@@ -0,0 +1,289 @@
+	const existingWebLPAQuestionnaireReviewOutcome = request.session.webLPAQuestionnaireReviewOutcome;
+
+	if (
+		existingWebLPAQuestionnaireReviewOutcome &&
+		(existingWebLPAQuestionnaireReviewOutcome.appealId !== appealId ||
+			existingWebLPAQuestionnaireReviewOutcome.lpaQuestionnaireId !== lpaQuestionnaireId ||
+			existingWebLPAQuestionnaireReviewOutcome.validationOutcome !== 'incomplete')
+	) {
+		// @ts-ignore
+		delete request.session.webLPAQuestionnaireReviewOutcome;
+	}
+
+	// @ts-ignore
+	const { webLPAQuestionnaireReviewOutcome } = request.session;
+	const incompleteReasonOptions = await getLPAQuestionnaireIncompleteReasons();
+
+	if (incompleteReasonOptions) {
+		const incompleteReasons =
+			body.incompleteReason || webLPAQuestionnaireReviewOutcome?.incompleteReasons;
+		const otherReason = body.otherReason || webLPAQuestionnaireReviewOutcome?.otherReason;
+		const mappedIncompleteReasonOptions = mapIncompleteReasonsToCheckboxItemParameters(
+			incompleteReasonOptions,
+			incompleteReasons
+		);
+
+		const appealReferenceFragments = appealReference.split('/');
+
+		return response.render('appeals/appeal/lpa-questionnaire-incomplete.njk', {
+			appeal: {
+				id: appealId,
+				shortReference: appealReferenceFragments?.[appealReferenceFragments.length - 1]
+			},
+			lpaQuestionnaireId,
+			reasonOptions: mappedIncompleteReasonOptions,
+			otherReasonId: getIdByNameFromIdNamePairs(incompleteReasonOptions, 'other'),
+			otherReason,
+			errors
+		});
+	}
+
+	return response.render('app/500.njk');
+};
+
+/**
+ *
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+const renderUpdateDueDate = async (request, response) => {
+	const { errors } = request;
+
+	if (
+		!objectContainsAllKeys(request.session, ['appealId', 'appealReference', 'lpaQuestionnaireId'])
+	) {
+		return response.render('app/500.njk');
+	}
+
+	const {
+		// @ts-ignore
+		appealId,
+		// @ts-ignore
+		appealReference,
+		// @ts-ignore
+		lpaQuestionnaireId
+	} = request.session;
+
+	const appealReferenceFragments = appealReference.split('/');
+
+	return response.render('appeals/appeal/update-due-date.njk', {
+		appeal: {
+			id: appealId,
+			shortReference: appealReferenceFragments?.[appealReferenceFragments.length - 1]
+		},
+		page: {
+			title: 'Appellant case due date',
+			text: 'Update LPA questionnaire due date'
+		},
+		backButtonUrl: `/appeals-service/appeal-details/${appealId}/lpa-questionnaire/${lpaQuestionnaireId}`,
+		skipButtonUrl: `/appeals-service/appeal-details/${appealId}/lpa-questionnaire/${lpaQuestionnaireId}/check-your-answers`,
+		errors
+	});
+};
+
+/**
+ *
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ * @param {string} appealShortReference
+ * @param {number} appealId
+ * @param {import("../../appellant-case/appellant-case.service.js").DayMonthYear} updatedDueDate
```

Comment:
Probably not one for this PR, but at some point we should move any types that are shared like this into a common file further up the directory structure, to avoid cross-referencing between features like this